src/args_parser.c

```diff
@@ -2,10 +2,22 @@
 
 #include <stdlib.h>
 
-args_t args_parse(char **argv) {
+args_t args_parse(int argc, char **argv) {
     args_t parsed;
+    parsed.flags[0].flag_name = "";
+    parsed.flags[0].flag_value = "";
+    parsed.flags[1].flag_name = "";
+    parsed.flags[1].flag_value = "";
 
-    
+
+    for(int i = 0; i < 4 && i < argc-1; i = i + 2)
+    {
+        parsed.flags[i].flag_name = argv[i];
+        if(i+1 < argc-1)
```

Comment:
You should probably keep the pattern on either `expr1 binOp expr2` or `expr1binOpexpr2`. E.g.:

` if(i + 1 < argc - 1)`, instead of ` if(i+1 < argc-1)`. Line 13 for example has no pattern. It hurts my eyes, I had to comment. xD

Reply:
Good catch!